pkg/util/tls_certificate.go

```diff
@@ -0,0 +1,72 @@
+package util
+
+import (
+	"bytes"
+	"crypto/tls"
+	"os"
+	"sync"
+)
+
+// RotatingTLSCertificate provides an up-to-date certificate given file paths and a refresh interval.
+type RotatingTLSCertificate struct {
+	sync.RWMutex
```

Comment:
Nit: Could you make this RWMutex named? There is no need for this type to offer a `Lock()`/`Unlock()` as part of the public API, right?

Furthermore, general convention in the Buildbarn tree is to group struct elements by locking requirements. So maybe best to write it like this:
```go
type RotatingTLSCertificate struct {
    certFile string
    keyFile string

    lock sync.RWMutex
    cert *tls.Certificate
    certData []byte
    keyData []byte
}
```
This makes it more obvious that certFile and keyFile are immutable.

Reply:
Done! Thanks for the tip :)